vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridScrollToIT.java

```diff
@@ -89,4 +89,40 @@ public void grid_addItem_scrollToIndex() {
                 grid.getCell(grid.getLastVisibleRowIndex(), 0).getText());
     }
 
+    @Test
+    public void grid_addItem_scrollToIndex_twice() {
+        $("button").id("add-row-and-scroll-to-index").click();
+        // Wait until finished loading
+        waitUntil(e -> !grid.getRow(grid.getFirstVisibleRowIndex())
+                .hasAttribute("loading"));
+
+        $("button").id("add-row-and-scroll-to-index").click();
+        waitUntil(e -> !grid.getRow(grid.getFirstVisibleRowIndex())
+                .hasAttribute("loading"));
+
+        // Find the content element of the first visible row cell
+        var slot = grid.getCell(grid.getFirstVisibleRowIndex(), 0)
+                .findElement(By.tagName("slot"));
+        var content = grid
+                .findElement(By.cssSelector("vaadin-grid-cell-content[slot='"
+                        + slot.getPropertyString("name") + "']"));
+        // Expect the content element to be displayed
+        Assert.assertTrue(content.isDisplayed());
```

Comment:
I'm a bit confused what this test actually does. If the cell content is displayed (I assume it means no display: none), does it gurantee that the content is actually there? Would it make sense to check for the cell text instead?

Reply:
This also checks `visibility: hidden` which is applied to the content in the case a row `<tr>` is in a loading state:

![Screenshot 2023-09-18 at 13 16 47](https://github.com/vaadin/flow-components/assets/1222264/c4ad1948-d0b5-48c5-962b-c6106ad2be37)

Checking for the text would not be meaningful in the case of the error (missing changes to the connector) because the text content would actually be correct. The problem is that the content is invisible due to the row being stuck in a loading-state.

Reply:
Ah, I see. Would it be enough to just wait for the row to exit the loading state then, similarly to the test below?

Reply:
Technically, it would be enough, but I kind of considered checking the `loading` attribute of an element inside the grid's shadow root as an internal detail, whereas the real user-facing symptom of the issue is the content not being displayed.